Clock/main.qml

```diff
@@ -48,97 +60,315 @@ Window {
+                    }
+                }
+            }
+
+            AlarmDateButton {
+                id: alarmDate
+
+                anchors.horizontalCenter: alarm.horizontalCenter
+                anchors.horizontalCenterOffset: width/2+10
+
+                state: "disabled"
+                buttonTxt: "Set date"
+
+                MouseArea {
+                    anchors.fill: parent
+                    onClicked: {
+                        parent.state="selected-hover"
+                        transitionTimer.thenChange=parent
+                        transitionTimer.nextView="Set date"
+                        transitionTimer.nextState= "selected"
+                        transitionTimer.running=true
+                        alarmEveryday.state= "disabled"
+                        alarm.everyDay= false
+                    }
+                }
+            }
+
+            AlarmSetHour {
+                id: alarmSetHour
+
+                isActive: root.view==="Alarm"
+            }
+
+            SetButton {// set date
+
+                state: "disable"
+                buttonTxt: "SET ALARM"
+
+                MouseArea {
+                    anchors.fill: parent
+                    onClicked: {
+                        if(!alarm.everyDay&&!alarm.setDate) return
+
+                        var appendDate=alarmSetHour.hourSet
+
+                        if(alarm.everyDay)
+                            alarmList.everyDayAlarms.append({
+                                "date": appendDate,
+                                "everyDay": true,
+                                "isActive": true
+                            })
+                        else{
+                            appendDate.setDate(
+                                alarmSetDate.newDate.getDate())
+                            appendDate.setMonth(
+                                        alarmSetDate.newDate.getMonth())
+                            appendDate.setFullYear(
+                                        alarmSetDate.newDate.getFullYear())
+                            alarmList.dateAlarms.append({
+                                "date": appendDate,
+                                "everyDay": false,
+                                "isActive": true,
+                                })
+                        }
+
+                        parent.state= "active"
+                        transitionTimer.thenChange=parent
+                        transitionTimer.nextView="DeveClock"
+                        transitionTimer.nextState="disable"
+                        transitionTimer.running=true
+                        isThereAlarm=true
+                    }
                 }
             }
         }
 
-        AlarmDateButton{
-            id: alarmDay
-            state: "selected"
-            buttonTxt: "Set date"
-            x: 254
+
+        //-------------------------------- view Alarm clock list
+        Item {
```

Comment:
@franchiStud cerca di riportare tutte queste pagine in file separati, come suggerito nella traccia dell'esercizio. Ti dà una struttura migliore per l'applicazione.